tests/pytests/unit/modules/test_suse_ip.py

```diff
@@ -0,0 +1,694 @@
+"""
+    :codeauthor: Jayesh Kariya <jayeshk@saltstack.com>
+"""
+
+import copy
+import os
+
+import jinja2.exceptions
+import pytest
+import salt.modules.suse_ip as suse_ip
+from tests.support.mock import MagicMock, patch
+
+
+@pytest.fixture
+def configure_loader_modules():
+    return {suse_ip: {"__grains__": {"os_family": "Suse"}}}
+
+
+def test_error_message_iface_should_process_non_str_expected():
+    values = [1, True, False, "no-kaboom"]
+    iface = "ethtest"
+    option = "test"
+    msg = suse_ip._error_msg_iface(iface, option, values)
+    assert msg
+    assert msg.endswith("[1|True|False|no-kaboom]")
+
+
+def test_error_message_network_should_process_non_str_expected():
+    values = [1, True, False, "no-kaboom"]
+    msg = suse_ip._error_msg_network("fnord", values)
+    assert msg
+    assert msg.endswith("[1|True|False|no-kaboom]")
+
+
+def test_build_interface():
+    """
+    Test to build an interface script for a network interface.
+    """
+    with patch.object(suse_ip, "_raise_error_iface", return_value=None):
+        with pytest.raises(AttributeError):
+            suse_ip.build_interface("iface", "slave", True)
+
+        with patch.dict(
+            suse_ip.__salt__, {"network.interfaces": lambda: {"eth": True}}
+        ), pytest.raises(AttributeError):
+            suse_ip.build_interface(
+                "iface",
+                "eth",
+                True,
+                netmask="255.255.255.255",
+                prefix=32,
+                test=True,
+            )
+            suse_ip.build_interface(
+                "iface",
+                "eth",
+                True,
+                ipaddrs=["A"],
+                test=True,
+            )
+            suse_ip.build_interface(
+                "iface",
+                "eth",
+                True,
+                ipv6addrs=["A"],
+                test=True,
+            )
```

Comment:
Are we sure the 3 `suse_ip.build_interface` calls get called, don't we really need one `with pytest.raises(AttributeError)` per call?

Reply:
Not sure if I understood the question, but what is the difference?